pkg/controllermanager/controller/seed/seed_backup_bucket_check_control.go

```diff
@@ -57,41 +58,46 @@ func (c *Controller) backupBucketUpdate(oldObj, newObj interface{}) {
 		return
 	}
 
-	if !apiequality.Semantic.DeepEqual(oldBackupBucket.Status, newBackupBucket.Status) || !apiequality.Semantic.DeepEqual(oldBackupBucket.Spec, newBackupBucket.Spec) {
+	if lastErrorChanged(oldBackupBucket.Status.LastError, newBackupBucket.Status.LastError) {
 		c.backupBucketEnqueue(newBackupBucket)
 	}
 }
 
-// NewBackupBucketReconciler returns a new default control to checks backup buckets of related seeds.
-func NewBackupBucketReconciler(gardenClient client.Client) *backupBucketReconciler {
-	return &backupBucketReconciler{
+// NewBackupBucketsCheckReconciler creates a new reconciler that maintains the BackupBucketsReady condition of Seeds
+// according to the observed status of BackupBuckets.
+func NewBackupBucketsCheckReconciler(gardenClient client.Client, config config.SeedBackupBucketsCheckControllerConfiguration, clock clock.Clock) *backupBucketsCheckReconciler {
+	return &backupBucketsCheckReconciler{
 		gardenClient: gardenClient,
+		config:       config,
+		clock:        clock,
 	}
 }
 
-type backupBucketReconciler struct {
+type backupBucketsCheckReconciler struct {
 	gardenClient client.Client
+	config       config.SeedBackupBucketsCheckControllerConfiguration
+	clock        clock.Clock
 }
 
 type backupBucketInfo struct {
 	name     string
 	errorMsg string
 }
 
-func (b *backupBucketInfo) String() string {
+func (b backupBucketInfo) String() string {
 	return fmt.Sprintf("Name: %s, Error: %s", b.name, b.errorMsg)
 }
 
-func (b *backupBucketReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+func (b *backupBucketsCheckReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
 	log := logf.FromContext(ctx)
 
 	seed := &gardencorev1beta1.Seed{}
 	if err := b.gardenClient.Get(ctx, req.NamespacedName, seed); err != nil {
 		if apierrors.IsNotFound(err) {
 			log.V(1).Info("Object is gone, stop reconciling")
-			return reconcile.Result{}, nil
+			return reconcileResult(nil)
```

Comment:
Can you drop this `reconcileResult` function and revert related changes? We don't have this anywhere else and it's unnecessarily "simplifying" the code